client/src/pages/SpaceDetail/Activity/Activity.tsx

```diff
@@ -82,9 +82,16 @@ export const Activity: React.FC<Props> = ({ spaceId, space }) => {
           Activity
         </Typography>
 
-        {transactions.length === 0 && (
+        {txQuery.isLoading ? (
+          <div className="flex justify-center items-center py-12">
+            <div className="text-center">
+              <Spinner color="blue" className="h-8 w-8 mx-auto mb-4" />
+              <p className="text-gray-600">Loading transactions... =^ω^=</p>
+            </div>
+          </div>
+        ) : transactions.length === 0 ? (
           <div className="text-gray-700">No transactions yet.</div>
-        )}
+        ) : null}
```

Comment:
I think to simplify the conditions here, a solution is to put the loading check inside the block of `transactions.length === 0`. 

These loaders look exactly the same, same class names, similar wrapper, can we have the loader into a separate component and call it everywhere? I think that'd be less confusing